sharktank/sharktank/types/tensors.py

```diff
@@ -752,10 +752,29 @@ class ShardedTensor(InferenceTensor):
     """
 
     def __init__(
-        self, *, shape: list[int], shard_dim: int | None, name: str = UnnamedTensorName
+        self, *, 
+        shape: list[int], 
+        shard_dim: int | None, 
+        name: str = UnnamedTensorName,
+        devices: Tuple[int],
```

Comment:
Devices should be `Optional[Tuple[int]] = None`. Then when not set we can initialize according to the number of shards.

Reply:
I think the `None` cases for `devices` and `devices_pinned` should be dealt with in `ShardedTensorBase.__init__`.

`ShardedTensor.__init__` is called from it before the shards are set, so we won't have access to the underlying `torch.Tensors` to infer anything.